mass/core/optimal_filtering.py

```diff
@@ -296,6 +299,126 @@ def report(self, std_energy: float = 5898.8):
         print(f"v/\u03b4v: {v_dv: .2f}, variance: {var:.2f} \u03b4E: {fwhm_eV:.2f} eV (FWHM), assuming standard E={std_energy:.2f} eV")
 
 
+@dataclass(frozen=True)
+class Filter5Lag(Filter):
+    """Represent an optimal filter, specifically one intended for 5-lag convolution with data
+
+        The traditional 5-lag filter used by default until 2015.
+
+    Returns
+    -------
+    Filter5Lag
+        An optimal filter, for convolution with data (at 5 lags, obviously)
+    """
+
+    def __post_init__(self):
+        assert self.convolution_lags == 5
+
+    @property
+    def _filter_type(self):
+        return "5lag"
+
+    # These parameters fit a parabola to any 5 evenly-spaced points
+    FIVELAG_FITTER = np.array((
+        (-6, 24, 34, 24, -6),
+        (-14, -7, 0, 7, 14),
+        (10, -5, -10, -5, 10),
+    ), dtype=float) / 70.0
+
+    def filter_records(self, x: npt.ArrayLike) -> tuple[np.ndarray, np.ndarray]:
+        """Filter one microcalorimeter record or an array of records.
+
+        Parameters
+        ----------
+        x : npt.ArrayLike
+            A 1-d array, a single pulse record, or a 2-d array, each row a pulse records.
```

Comment:
For those of us (me) who find it impossible to remember which index is a row, can we write "`x[i,:]` is the `i`th pulse."

Reply:
Yes, done (in next commit).